mlrun/api/db/sqldb/db.py

```diff
@@ -665,6 +673,211 @@ def delete_project(self, session, name: str):
+
+        feature_set = FeatureSet(
+            name=name,
+            project=project,
+            state=state,
+            status=feature_set_dict.get("status", {}),
+            uid=uid,
+            updated=datetime.now(timezone.utc),
+        )
+
+        feature_set_spec = feature_set_dict.get("spec")
+        if feature_set_spec:
+            features = feature_set_spec.pop("features", [])
+            entities = feature_set_spec.pop("entities", [])
+
+        self._update_feature_set_features(feature_set, features, is_entity=False)
+        self._update_feature_set_features(feature_set, entities, is_entity=True)
+
+        labels = feature_set_dict["metadata"].pop("labels", {})
+        update_labels(feature_set, labels)
+
+        self._upsert(session, feature_set)
+        self.tag_objects_v2(session, [feature_set], project, tag)
+
+        return uid
+
+    def update_feature_set(
+        self,
+        session,
+        project,
+        name,
+        feature_set_update: schemas.FeatureSetUpdate,
+        tag=None,
+        uid=None,
+    ):
+        # If object uid was given, use it. Else, find it by tag
+        if not uid:
+            tag = tag or "latest"
+            tag = self._query(
+                session, FeatureSet.Tag, project=project, name=tag, obj_name=name
+            ).one_or_none()
+            if not tag:
+                feature_set_uri = generate_object_uri(project, name, tag)
+                raise mlrun.errors.MLRunNotFoundError(
+                    f"Feature-set not found {feature_set_uri}"
+                )
+            query = self._query(session, FeatureSet, id=tag.obj_id)
+        else:
+            query = self._query(
+                session, FeatureSet, name=name, project=project, uid=uid
+            )
+
+        feature_set_record = query.one_or_none()
+        if not feature_set_record:
+            feature_set_uri = generate_object_uri(project, name, tag)
+            raise mlrun.errors.MLRunNotFoundError(
+                f"Feature-set not found {feature_set_uri}"
+            )
+
+        data = feature_set_update.dict()
+
+        # TODO (future) - granular update of JSON fields.
+        status = data.pop("status", None)
+        if status:
+            feature_set_record.status = status
+            feature_set_record.state = status.get("status") or feature_set_record.state
+
+        features = data.pop("features", [])
+        if features:
+            self._update_feature_set_features(
+                feature_set_record, features, is_entity=False, replace=True
+            )
+
+        entities = data.pop("entities", [])
+        if entities:
+            self._update_feature_set_features(
+                feature_set_record, entities, is_entity=True, replace=True
+            )
+
+        merge_labels(feature_set_record, data.pop("labels", {}))
+        feature_set_record.updated = datetime.now(timezone.utc)
+
+        self._upsert(session, feature_set_record, ignore=True)
```

Comment:
Why ignore ?

Reply:
hmm... Trying to remember why, can't really find a good reason. Maybe it was a testing thing that I forgot to remove. 
Fixed.